src/ProxyProtocol.cc

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (C) 1996-2018 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include <algorithm>
+#include "parser/BinaryTokenizer.h"
+#include "parser/Tokenizer.h"
+#include "ProxyProtocol.h"
+#include "sbuf/StringConvert.h"
+#include "sbuf/Stream.h"
+#include "SquidConfig.h"
+#include "SquidString.h"
+#include "StrList.h"
+
+#if HAVE_SYS_SOCKET_H
+#include <sys/socket.h>
+#endif
+#if HAVE_NETINET_IN_H
+#include <netinet/in.h>
+#endif
+#if HAVE_NETINET_IP_H
+#include <netinet/ip.h>
+#endif
+
+/// magic octet prefix for PROXY protocol version 1
+static const SBuf Proxy1p0magic("PROXY ", 6);
```

Comment:
Avoid Foo inside Foo and bare "proxy". Here is a sketch that will also solve a few other, similar problems:

```suggestion
namespace ProxyProtocol {
    namespace One {
        static const SBuf Magic("PROXY ", 6);
        static Message::Pointer Parse(::Parser::Tokenizer &tok);
    }
    namespace Two {
        static const SBuf Magic("\x0D\x0A\x0D\x0A\x00\x0D\x0A\x51\x55\x49\x54\x0A", 12);
        static Message::Pointer Parse(::Parser::Tokenizer &tok);
    }
}
```